### ISSUE TITLE
How to disable location detection?

### ISSUE THREAD

**Issue by gokanil** [ultrafunkamsterdam/undetected-chromedriver#686]
When I activate VPN, some sites detect that it is in another location. How can I solve this?

**Comment by sebdelsol** [comment]
You can't prevent a server to [geolocate your IP](https://geotargetly.com/my-ip-geolocation). 
BTW using a VPN is often a bad idea since they tend to be flagged as suspicious by a lot of sites.

I'm not sure about your question though : 
Do you mean some sites detect a discrepancy between you IP location and your position given by chrome location service ? 
In that case you'll want to disable chrome location services:  
```python3
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait

import undetected_chromedriver as uc

if __name__ == "__main__":
    options = uc.ChromeOptions()
    prefs = {"profile.default_content_settings.geolocation": 2}
    options.add_experimental_option("prefs", prefs)
    options.add_argument("--deny-permission-prompts")
    driver = uc.Chrome(options=options)

    # will show : "Sorry, Google Maps does not have permission to use your location."
    driver.get("https://www.labnol.org/maps/location/")
    button_xpath = By.XPATH, "//button[contains(text(),'Location')]"
    WebDriverWait(driver, 5).until(EC.element_to_be_clickable(button_xpath)).click()
    input("hit Enter to quit")
```

**Comment by gokanil** [comment]
"profile.default_content_settings.geolocation": 2

After making these settings, I still get an same error. I will not type the url address because I don't know if it is correct to type the url address. But here's the error I got in the console output:

"LocationNotAllowedException: User is not allowed to use the app in their current location."

**Comment by sebdelsol** [comment]
So you want to **enable** the location : 
* `prefs = {"profile.default_content_settings.geolocation": 1}`
* and remove `options.add_argument("--deny-permission-prompts")`

**Comment by gokanil** [comment]
No, I don't want this. What I want is: After connecting to VPN, it does not open the website because it detects that I am in a different location. It is not enough to change the geolocation settings.

**Comment by sebdelsol** [comment]
so there’s a discrepancy between your IP location and something else that hints at your actual location… have you tried to change your Chrome language to reflect your spoofed location ?

**Comment by gokanil** [comment]
My browser language is English by default. I connected to the US-state of California for the test and the result is the same. How else could they know?

**Comment by sebdelsol** [comment]
I don't know, do you have a very **_simple_** use case to trigger your issue ?

**Comment by gokanil** [comment]
Before writing the address, I would like to mention this. My goal is not a trick or piracy. All I want is to see Selenium as undetected.
[link](https://www.disneyplus.com/en-gb/login)

**Comment by hulitolku** [comment]
> So you want to **enable** the location :
> 
> * `prefs = {"profile.default_content_settings.geolocation": 1}`
> * and remove `options.add_argument("--deny-permission-prompts")`
Dont work.

**Comment by sebdelsol** [comment]
@hulitolku , what do you mean it doesn't work ?

**Comment by hulitolku** [comment]
there is no click on the allow button

**Comment by sebdelsol** [comment]
I still don't understand, could you please show some code and what's you're trying to achieve ?

**Comment by hulitolku** [comment]
I need clik "Allow"
[https://i.stack.imgur.com/qoWFn.png](url)

**Comment by sebdelsol** [comment]
You don't need to click "allow" if it's already allowed ?

**Comment by hulitolku** [comment]
> You don't need to click "allow" if it's already allowed ?

"already allowed" not work if i use `options.add_argument('--user-data-dir=c:\\temp\\profile1')`
